### PR TITLE
[PIR] Migrate paddle.nn.Sigmoid into pir

### PR DESCRIPTION
**Pull request by MarioLulab**
### PR types
Others

### PR changes
Others

### Description
新 IR API 推全升级
将 `paddle.nn.Sigmoid` 迁移升级至 new ir，并更新单测 本 pr 进行补充.
`paddle.nn.Sigmoid` 中包含 `paddle.nn.functional.sigmoid` 调用，故一起进行适配

单测覆盖率：8/14

备注：
1. `test/legacy_test/test_nn_sigmoid_op.py` 内的静态图测试涉及 `base.backward.append_backward` 函数，当前版本会报错：
  
  ```
  The type of 'loss' in paddle.static.append_backward must be <class 'paddle.base.framework.Variable'>, but received <class 'paddle.base.libpaddle.pir.OpResult'>. 
  ```

2. 当前版本反向传播机制对 pir api 适配还不完善，待完善后会继续补充与反向传播相关的单测

3. 部分静态图单测未切换到 pir 体系下，后续需补充

4. 缺失 `TestMKLDNNSigmoidBF16Op` 单测

### CODE REVIEW

**Review by Aurelius84 — APPROVED** [review]
LGTM
